### PR TITLE
Adding Non-equilibrium Green's Function module into postprocessing of DeePTB

### LINKED ISSUE #31: Adding Non-equilibrium Green's Function module into postprocessing of DeePTB
### Describe Current Status and Possible Solution

An important application of the Tight-Binding Model is to simulate transport phenomena with Non-equilibrium Green's Function NEGF methods. This module was determined to enable DeePTB to model electronic devices with multiple electrodes and simulate transport properties accurately. Basic sub-modules are:

1. The NEGF sub-module, which performs the NEGF calculations;
2. The Lead and Device structure modules, which support modelling Devices with scattering regions that are concatenated with several electrodes. Both periodic systems and confined systems are supported.
3. The interfaces between the NEGF module and the Postprocessing module of DeePTB.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by floatingCatty**
Currently, the modules contain:

1. NEGF module, which supports the computation of surface green's function, recursive green's function, Ozaki contour integration for density, PDIIS self-consistent iteration.
2. NEGF's interface with DeePTB run module.
3. Device and Lead structure class to build two-end devices.

Fix #31

### CODE REVIEW

**Comment by floatingCatty** [comment]
The test failed because the newly introduced tests and modules require updating the dependency list, which is not compiled now.

File: dptb/negf/Areshkin.py
Comment (QG-phy) [inline dptb/negf/Areshkin.py:1]: 函数名的命名规范, 参考：
https://zh-google-styleguide.readthedocs.io/en/latest/google-python-styleguide/python_style_rules/#section-9

eg:

模块名: module_name; 包名: package_name; 类名: ClassName; 方法名: method_name; 异常名: ExceptionName; 函数名: function_name, query_proper_noun_for_thing, send_acronym_via_https; 全局常量名: GLOBAL_CONSTANT_NAME ; 全局变量名: global_var_name; 实例名: instance_var_name; 函数参数名: function_parameter_name; 局部变量名: local_var_name.

File: dptb/negf/CFR.py
Comment (QG-phy) [inline dptb/negf/CFR.py:1]: CFR，本身作为函数名，非常的没有识别性，看名字不知道这个是什么。改一下吧？还有注意命名规范。

File: dptb/negf/RGF.py
Comment (QG-phy) [inline dptb/negf/RGF.py:1]: 命名规范，还有名字的辨识度。RGF 看名字很难知道这个是什么。

File: dptb/negf/hamiltonian.py
Comment (QG-phy) [inline dptb/negf/hamiltonian.py:1]: hamiltonian 怎么？具体是干嘛？device hamiltonian ? 函数名的与我们dptb.hamiltonian 是重复的，这个在import的时候容易误解。修改一下吧。

File: dptb/negf/sgf.py
Comment (QG-phy) [inline dptb/negf/sgf.py:1]: sgf 代表了什么？

File: dptb/negf/utils.py
Comment (QG-phy) [inline dptb/negf/utils.py:1]: 这个utils 函数与我们的 dptb.utils 模块重名，容易误解修改下吧。

**Review by QG-phy — APPROVED** [review]
This PR is a basic version NEGF code, only supports two leads and the Homojunction device.

I reviewed this PR, and found it follows the rules for contributing to DeePTB.  I approve this PR. 

But developing NEGF still needs a lot of development.

**Pull request merged**